code/modules/unit_tests/holidays.dm

```diff
@@ -0,0 +1,35 @@
+// test Jewish holiday
+/datum/unit_test/hanukkah_2123/Run()
+	var/datum/holiday/hebrew/hanukkah/hanukkah = new
+	TEST_ASSERT_EQUAL(hanukkah.shouldCelebrate(13, DECEMBER, 2123, 2, MONDAY), TRUE, "December 13, 2123 was not Hannukkah.")
+
+// test Islamic holiday
+/datum/unit_test/ramadan_4208/Run()
+	var/datum/holiday/islamic/ramadan/ramadan = new
+	TEST_ASSERT_EQUAL(ramadan.shouldCelebrate(1, MARCH, 4208, 1, TUESDAY), TRUE, "March 1, 4208 was not the start of Ramadan.")
+
+// there were problems on this day
+/datum/unit_test/nov_1_2018/Run()
+	for(var/H in subtypesof(/datum/holiday))
+		var/datum/holiday/holiday = new H()
+		TEST_ASSERT_EQUAL(holiday.shouldCelebrate(1, NOVEMBER, 2018, 1, TUESDAY), FALSE, "A holiday tried to celebrate on November 1, 2018.")
```

Comment:
This is a very strange, and non forwards compatible test! What's the deal with November 1st, 2018, and why should no *future* holidays be allowed to be celebrated on that day?

Reply:
True